packages/salesforcedx-vscode-core/src/commands/forceProjectCreate.ts

```diff
@@ -31,14 +31,32 @@ import {
   SfdxCommandletExecutor
 } from './commands';
 
-export class ForceProjectCreateExecutor extends SfdxCommandletExecutor<{}> {
+type forceProjectCreateOptions = {
+  isChangeSetBasedProject: boolean;
+};
+
+export class ForceProjectCreateExecutor extends SfdxCommandletExecutor<
+  ProjectNameAndPath
+> {
+  private readonly options: forceProjectCreateOptions;
+
+  public constructor(options = { isChangeSetBasedProject: false }) {
+    super();
+    this.options = options;
+  }
+
   public build(data: ProjectNameAndPath): Command {
-    return new SfdxCommandBuilder()
+    const builder = new SfdxCommandBuilder()
       .withDescription(nls.localize('force_project_create_text'))
       .withArg('force:project:create')
       .withFlag('--projectname', data.projectName)
-      .withFlag('--outputdir', data.projectUri)
-      .build();
+      .withFlag('--outputdir', data.projectUri);
+
+    if (this.options.isChangeSetBasedProject) {
```

Comment:
Main change is here. This provides the option to also pass the `--manifest` flag.